src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -2429,4 +2442,345 @@ public Response getTimestamps(@PathParam("identifier") String id) {
+            DatasetLock dcmLock = dataset.getLockFor(DatasetLock.Reason.EditInProgress);
+            if (dcmLock == null) {
+                logger.log(Level.WARNING, "No lock found for dataset");
+            } else {
+                datasetService.removeDatasetLocks(dataset, DatasetLock.Reason.EditInProgress);
+                logger.log(Level.INFO, "Removed EditInProgress lock ");
+                //dataset.removeLock(dcmLock);
+            }
+
+            try {
+                Command<Dataset> cmd;
+                cmd = new UpdateDatasetVersionCommand(dataset, dvRequest);
+                ((UpdateDatasetVersionCommand) cmd).setValidateLenient(true);
+                commandEngine.submit(cmd);
+            } catch (CommandException ex) {
+                logger.log(Level.WARNING, "==== datasetId :" + dataset.getId() + "====== UpdateDatasetVersionCommand Exception  : " + ex.getMessage());
+            }
+
+            dataset = datasetService.find(dataset.getId());
+
+            List<DataFile> s = dataset.getFiles();
+            for (DataFile dataFile : s) {
+            }
+            //ingest job
+            ingestService.startIngestJobsForDataset(dataset, (AuthenticatedUser) authUser);
+
+        } catch (Exception e) {
+            String message = e.getMessage();
+            msgt("*******  datasetId :" + dataset.getId() + " ======= addFilesToDataset CALL Exception ============== " + message);
+            e.printStackTrace();
+        }
+
+        return ok(Json.createObjectBuilder().add("Files", jarr));
+
+    } // end: addFileToDataset
+
+
+    @POST
+    @Path("{id}/deleteglobusRule")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response deleteglobusRule(@PathParam("id") String datasetId,@FormDataParam("jsonData") String jsonData
+    ) throws IOException, ExecutionException, InterruptedException {
+
+
+        logger.info(" ====  (api deleteglobusRule) jsonData   ====== " + jsonData);
+
+
+        if (!systemConfig.isHTTPUpload()) {
+            return error(Response.Status.SERVICE_UNAVAILABLE, BundleUtil.getStringFromBundle("file.api.httpDisabled"));
+        }
+
+        // -------------------------------------
+        // (1) Get the user from the API key
+        // -------------------------------------
+        User authUser;
+        try {
+            authUser = findUserOrDie();
+        } catch (WrappedResponse ex) {
+            return error(Response.Status.FORBIDDEN, BundleUtil.getStringFromBundle("file.addreplace.error.auth")
+            );
+        }
+
+        // -------------------------------------
+        // (2) Get the Dataset Id
+        // -------------------------------------
+        Dataset dataset;
+
+        try {
+            dataset = findDatasetOrDie(datasetId);
+        } catch (WrappedResponse wr) {
+            return wr.getResponse();
+        }
+
+        // Async Call
+        datasetService.globusDownload(jsonData, dataset, authUser);
```

Comment:
Should check whether this user should have download perms before going async so you can return 403?